core/commands/add.go

```diff
@@ -90,6 +94,17 @@ remains to be implemented.
 
 		progress, _, _ := req.Option(progressOptionName).Bool()
 		wrap, _, _ := req.Option(wrapOptionName).Bool()
+		hidden, _, _ := req.Option(hiddenOptionName).Bool()
+
+		var ignoreFilePatterns []ignore.GitIgnore
+
+		// Check the $IPFS_PATH
+		if ipfs_path := os.Getenv("$IPFS_PATH"); len(ipfs_path) > 0 {
+			baseFilePattern, err := ignore.CompileIgnoreFile(path.Join(ipfs_path, ".ipfsignore"))
+			if err == nil && baseFilePattern != nil {
+				ignoreFilePatterns = append(ignoreFilePatterns, *baseFilePattern)
+			}
+		}
```

Comment:
for env vars, want: 
```go
os.Getenv("IPFS_PATH") // no $
```

and i think we shouldn't be using the variable directly here-- we should be using the repo's path:

```go
req.Context().ConfigRoot
// we should perhaps have a way of getting this path from a .Repo.
```
It's also possible that we may want to pull it out of the repo with a function:

```go
// maybe not this (may want it as part of config?), but something equivalent to:
ignoreFilePatterns := Repo.IgnoreFiles()
```

the point being that repos which are not fsrepos (i.e. not in the filesystem) could still implement the functionality.

cc @whyrusleeping @tv42 @wking